### PR TITLE
feat(TKC-1652): persist TestWorkflow Executions in the database

### PR DESCRIPTION
**Pull request by rangoo94**
## Pull request description 

* Persist TestWorkflowExecution along with its result in the MongoDB
* Persist Log Output in Minio/S3
* Add CLI commands for all the execution operations
* Add API for all of the execution operations
* Fix some bugs with the TestWorkflow controller

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [x] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Fixes

- https://linear.app/kubeshop/issue/TKC-1652/testworkflows-prepare-persistence-for-the-executions

### CODE REVIEW

File: pkg/tcl/testworkflowstcl/testworkflowcontroller/logs.go [hunk 1]
```diff
@@ -178,6 +178,7 @@ func WatchContainerLogs(ctx context.Context, clientSet kubernetes.Interface, pod
 		err := WaitUntilContainerIsStarted(ctx, podEvents, containerName)
 		if err != nil {
 			w.SendError(err)
+			fmt.Println("waiting until started failed")
```
Comment (vsukhin) [inline pkg/tcl/testworkflowstcl/testworkflowcontroller/logs.go:181]: is this devugging?

File: pkg/storage/minio/minio.go [hunk 2]
```diff
@@ -632,3 +633,33 @@ func (c *Client) IsConnectionPossible(ctx context.Context) (bool, error) {
...
+
+func (c *Client) PresignDownloadFileFromBucket(ctx context.Context, bucket, bucketFolder, file string, expires time.Duration) (string, error) {
+	if err := c.Connect(); err != nil {
+		return "", err
+	}
+	if bucketFolder != "" {
+		file = strings.Trim(bucketFolder, "/") + "/" + file
+	}
+	c.Log.Debugw("presigning get object from minio", "file", file, "bucket", bucket)
+	url, err := c.minioClient.PresignedPutObject(ctx, bucket, file, expires)
```
Comment (vsukhin) [inline pkg/storage/minio/minio.go:645]: will this work for cloud agent?

File: pkg/tcl/apitcl/v1/testworkflows.go [hunk 3]
```diff
@@ -70,10 +68,13 @@ func (s *apiTCL) DeleteTestWorkflowHandler() fiber.Handler {
 		if err != nil {
 			return s.ClientError(c, errPrefix, err)
 		}
-		//skipExecutions := c.Query("skipDeleteExecutions", "")
-		//if skipExecutions != "true" {
-		//	// TODO: Delete Executions
-		//}
+		skipExecutions := c.Query("skipDeleteExecutions", "")
+		if skipExecutions != "true" {
```
Comment (vsukhin) [inline pkg/tcl/apitcl/v1/testworkflows.go:72]: should we chekc for bool ?

File: pkg/tcl/apitcl/v1/testworkflows.go [hunk 4]
```diff
@@ -82,23 +83,34 @@ func (s *apiTCL) DeleteTestWorkflowsHandler() fiber.Handler {
...
-		//	}
-		//}
+		// Mark as deleted
+		for range workflows.Items {
+			s.Metrics.IncDeleteTestWorkflow(err)
+		}
+
+		// Delete the executions
+		skipExecutions := c.Query("skipDeleteExecutions", "")
+		if skipExecutions != "true" {
```
Comment (vsukhin) [inline pkg/tcl/apitcl/v1/testworkflows.go:104]: should we check for bool?

File: pkg/tcl/apitcl/v1/testworkflows.go [hunk 5]
```diff
@@ -305,11 +318,27 @@ func (s *apiTCL) ExecuteTestWorkflowHandler() fiber.Handler {
 			return s.BadRequest(c, errPrefix, "processing error", err)
 		}
 
-		now := time.Now()
+		// Load execution identifier data
+		// TODO: Consider if that should not be shared (as now it is between Tests and Test Suites)
+		number, _ := s.ExecutionResults.GetNextExecutionNumber(context.Background(), workflow.Name)
+		executionName := request.Name
```
Comment (vsukhin) [inline pkg/tcl/apitcl/v1/testworkflows.go:324]: will be a bit confusing, if to use it together with tests

**Review by vsukhin — APPROVED** [review]
lgtm

File: pkg/tcl/apitcl/v1/testworkflows.go (hunk 5 above)
Comment (rangoo94) [reply pkg/tcl/apitcl/v1/testworkflows.go:324]: It is, but at the moment the `GetNextExecutionNumber` for tests and test suites is shared to - `sequences` do not have differentation between types.

File: pkg/tcl/apitcl/v1/testworkflows.go (hunk 4 above)
Comment (rangoo94) [reply pkg/tcl/apitcl/v1/testworkflows.go:104]: What do you mean?

File: pkg/storage/minio/minio.go (hunk 2 above)
Comment (vsukhin) [reply pkg/storage/minio/minio.go:645]: will we add cloud support later?

File: pkg/tcl/apitcl/v1/testworkflows.go (hunk 4 above)
Comment (rangoo94) [reply pkg/tcl/apitcl/v1/testworkflows.go:104]: `ParseBool` accepts multiple values as truthy - not sure if not better to be consistent with `true` only (like [**here**](https://github.com/kubeshop/testkube/blob/develop/internal/app/api/v1/tests.go#L505-L511))

File: pkg/storage/minio/minio.go (hunk 2 above)
Comment (rangoo94) [reply pkg/storage/minio/minio.go:645]: Yes, today. Only the Cloud will be supported initially.

File: pkg/tcl/apitcl/v1/testworkflows.go (hunk 4 above)
Comment (vsukhin) [reply pkg/tcl/apitcl/v1/testworkflows.go:104]: ok, let's follow existng approach

**Pull request merged**